core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -290,4 +301,22 @@ public void process(RestRequest request, RestChannel channel, RestFilterChain fi
             executeHandler(request, channel);
         }
     }
+
+    private static final class RestHandlerHolder {
```

Comment:
I also considered this but did not want to mix various aspects in the `RestHandler` interface. But I like the approach and can change that.